pipelines/controller/src/main/java/com/google/fhir/analytics/PipelineManager.java

```diff
@@ -166,10 +171,17 @@ private void setLastRunStatus(LastRunStatus status) {
   }
 
   @PostConstruct
-  private void initDwhStatus() {
+  private void initDwhStatus() throws ProfileMapperException {
 
     // Initialise the Flink configurations for all the pipelines
     initialiseFlinkConfiguration();
+    fhirContext =
+        ProfileMapperFhirContexts.getInstance()
+            .contextFor(dataProperties.getFhirVersion(), dataProperties.getProfileDefinitionsDir());
```

Comment:
Inside the next call for initializing ` avroConversionUtil`, we have another call of `ProfileMapperFhirContexts.contextFor` and they should be consistent; this is not great. I think under the `pipelines` module, there should be only one [singleton] class that is aware of ProfileMapperFhirContexts existence; everyone else should use that class for getting `FhirContext` (which is initialized with the right version and right profile-def-dir). That class can be `AvroConversionUtil` but regardless let's avoid this implicit consistency requirements (e.g., between `fhirContext` and `avroConversionUtil`) and enforce them in code.

Reply:
Yes this makes sense. I have made changes so that only `AvroConversionUtil` is aware of the `ProfileMapperFhirContexts` and everyone should use `AvroConversionUtil` instance to fetch `FhirContext`.